### PR TITLE
[WOC] Implement Korvold, Gleeful Glutton

### PR DESCRIPTION
**Pull request by xanderhall**
Submitting as a draft for now for preliminary review while I write unit tests

### CODE REVIEW

**Comment by xanderhall** [comment]
[[Korvold, Gleeful Glutton]]

File: Mage.Sets/src/mage/cards/k/KorvoldGleefulGlutton.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    instance;
+
+    @Override
+    public int calculate(Game game, Ability sourceAbility, Effect effect) {
+        KorvoldGleefulGluttonWatcher watcher = game.getState().getWatcher(null);
+        return watcher == null ? 0 : watcher.getNumberOfTypes(sourceAbility.getControllerId());
+    }
+
+    @Override
+    public DynamicValue copy() {
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/k/KorvoldGleefulGlutton.java:129]: covariant return here, not the super class. Also spelling "Sacrificed"

File: Mage.Sets/src/mage/cards/k/KorvoldGleefulGlutton.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    @Override
+    public int calculate(Game game, Ability sourceAbility, Effect effect) {
+        Player player = game.getPlayer(sourceAbility.getControllerId());
+        if (player == null) {
+            return 0;
+        }
+
+        return player.getGraveyard().getCards(game).stream()
+            .filter(card -> !card.isCopy() && !(card instanceof PermanentToken))
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/k/KorvoldGleefulGlutton.java:155]: curious how you'd end up with a copy or a token in the graveyard while evaluating this

File: Mage.Sets/src/mage/cards/k/KorvoldGleefulGlutton.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        return instance;
+    }
+
+    @Override
+    public String getMessage() {
+        return "card types among permanents you sacrificed";
+    }
+}
+
+enum PermanentTypesInGraveyardCount implements DynamicValue {
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/k/KorvoldGleefulGlutton.java:139]: I prefer naming card-specific classes with the card name so it's obvious they aren't common classes

File: Mage.Sets/src/mage/cards/k/KorvoldGleefulGlutton.java (hunk 2 above)
Comment (xanderhall) [reply Mage.Sets/src/mage/cards/k/KorvoldGleefulGlutton.java:155]: Copied that from `CardTypesInGraveyardCount`, assumed it was necessary

File: Mage.Sets/src/mage/cards/k/KorvoldGleefulGlutton.java (hunk 3 above)
Comment (xanderhall) [reply Mage.Sets/src/mage/cards/k/KorvoldGleefulGlutton.java:139]: I wanted to keep them clearly named, and I figured adding `KorvoldGleefulGlutton` to the beginning would make the names too wordy

**Review by xenohedron — APPROVED** [review]
